user.go

```diff
@@ -94,9 +105,24 @@ func (u *User) Mention() string {
 //    size:    The size of the user's avatar as a power of two
 //             if size is an empty string, no size parameter will
 //             be added to the URL.
+//
+// Update: https://discord.com/developers/docs/change-log#default-avatars
+// Based on the update from new username
+// Users on the legacy username will use the discriminator % 5
+// Otherwise, users will use the (user_id >> 22) % 6
 func (u *User) AvatarURL(size string) string {
-	return avatarURL(u.Avatar, EndpointDefaultUserAvatar(u.Discriminator),
-		EndpointUserAvatar(u.ID, u.Avatar), EndpointUserAvatarAnimated(u.ID, u.Avatar), size)
+	defaultUserAvatar := EndpointDefaultUserAvatar(u.Discriminator)
+	if u.isMigrated() {
+		defaultUserAvatar = EndpointDefaultUserAvatarMigrated(u.ID)
+	}
+
```

Comment:
Needs to be reworked, to include `index` calculations.

Reply:
And another one - `%6`, not `%5`.

Reply:
sorry for lil bit mistake, already fixed the modulo